packages/vite/src/node/build.ts

```diff
@@ -753,7 +753,21 @@ function prepareOutDir(
       config.publicDir &&
       fs.existsSync(config.publicDir)
     ) {
-      copyDir(config.publicDir, outDir)
+      if (areSeparateFolders(outDir, config.publicDir)) {
```

Comment:
Maybe this doesn't work for Windows, but with linux I think this can be computed as:
```js
config.publicDir.includes(`${outDir}/`) || outDir.includes(`${config.publicDir}/`)
```
(it's actually the opposite but that's the same goal)
This assume outDir is like publicDir: absolute wihtout trailing slash

Reply:
I think that won't work for dirs like `/a/b/c/d` and `/b/c`. It could be done with a regex though. I can try that to see if it is simpler.

Reply:
oh yeah but maybe startsWith should work?

Reply:
Done at [refactor: simplify areSeparateFolders](https://github.com/vitejs/vite/pull/13742/commits/2b9cc98b0643bc881f739ac141f6888a6cf840d2)